src/firefly/js/tables/TableUtil.js

```diff
@@ -761,7 +761,7 @@ export function processRequest(tableModel, tableRequest, hlRowIdx) {
     let {data, columns} = nTable.tableData;
 
     nTable.request = tableRequest;
-    pageSize = pageSize || data.length || MAX_ROW;
+    pageSize = pageSize > 0 ? pageSize : data.length || MAX_ROW;
```

Comment:
I think would be good to have a couple of tests for pagination in TableUtil-test.js like we already do there for sorting and filtering.

Reply:
test added.